src/Evolution/Systems/GeneralizedHarmonic/BoundaryConditionsImpl.hpp

```diff
@@ -0,0 +1,1447 @@
+                                   DataVector>::type& shift =
+        get<::Tags::TempI<20, VolumeDim, Frame::Inertial, DataVector>>(buffer);
+    const typename Tags::Pi<VolumeDim, Frame::Inertial>::type& pi =
+        get<Tags::Pi<VolumeDim, Frame::Inertial>>(vars);
+    const typename Tags::Phi<VolumeDim, Frame::Inertial>::type& phi =
+        get<Tags::Phi<VolumeDim, Frame::Inertial>>(vars);
+    const typename Tags::VSpacetimeMetric<VolumeDim, Frame::Inertial>::type&
+        char_projected_rhs_dt_u_psi =
+            get<::Tags::Tempaa<22, VolumeDim, Frame::Inertial, DataVector>>(
+                buffer);
+    const typename Tags::CharacteristicSpeeds<VolumeDim, Frame::Inertial>::type
+        char_speeds{{get(get<::Tags::TempScalar<12, DataVector>>(buffer)),
+                     get(get<::Tags::TempScalar<13, DataVector>>(buffer)),
+                     get(get<::Tags::TempScalar<14, DataVector>>(buffer)),
+                     get(get<::Tags::TempScalar<15, DataVector>>(buffer))}};
+
+    // Memory allocated for return type
+    ReturnType& bc_dt_u_psi =
+        get<::Tags::Tempaa<27, VolumeDim, Frame::Inertial, DataVector>>(buffer);
+    std::fill(bc_dt_u_psi.begin(), bc_dt_u_psi.end(), 0.);
+    // Switch on prescribed boundary condition method
+    switch (Method) {
+      case VSpacetimeMetricBcMethod::Freezing:
+        return bc_dt_u_psi;
+      case VSpacetimeMetricBcMethod::ConstraintPreservingBjorhus:
+        return apply_bjorhus_constraint_preserving(
+            make_not_null(&bc_dt_u_psi), unit_interface_normal_vector,
+            three_index_constraint, char_projected_rhs_dt_u_psi, char_speeds);
+      case VSpacetimeMetricBcMethod::ConstraintPreservingDirichlet:
+        return apply_dirichlet_constraint_preserving(
+            make_not_null(&bc_dt_u_psi), lapse, shift, pi, phi);
+      case VSpacetimeMetricBcMethod::Unknown:
+      default:
+        ASSERT(false,
+               "Requested BC method fo VSpacetimeMetric not implemented!");
+    }
+    // dummy return to suppress compiler warning
+    return ReturnType{};
+  }
+
+ private:
+  static ReturnType apply_bjorhus_constraint_preserving(
+      gsl::not_null<ReturnType*> bc_dt_u_psi,
+      const tnsr::A<DataVector, VolumeDim, Frame::Inertial>&
+          unit_interface_normal_vector,
+      const tnsr::iaa<DataVector, VolumeDim, Frame::Inertial>&
+          three_index_constraint,
+      const tnsr::aa<DataVector, VolumeDim, Frame::Inertial>&
+          char_projected_rhs_dt_u_psi,
+      const std::array<DataVector, 4>& char_speeds) noexcept;
+  static ReturnType apply_dirichlet_constraint_preserving(
+      gsl::not_null<ReturnType*> bc_dt_u_psi, const Scalar<DataVector>& lapse,
+      const tnsr::I<DataVector, VolumeDim, Frame::Inertial>& shift,
+      const tnsr::aa<DataVector, VolumeDim, Frame::Inertial>& pi,
+      const tnsr::iaa<DataVector, VolumeDim, Frame::Inertial>& phi) noexcept;
+};
+
+template <typename ReturnType, size_t VolumeDim>
```

Comment:
I think each of these needs doxygen help with the equations being implemented, with a citation to the reference and equation number where it comes from